server/src/main/java/org/opensearch/search/aggregations/bucket/terms/GlobalOrdinalsStringTermsAggregator.java

```diff
@@ -136,16 +144,92 @@ public GlobalOrdinalsStringTermsAggregator(
                 return new DenseGlobalOrds();
             });
         }
+        this.fieldName = (valuesSource instanceof ValuesSource.Bytes.WithOrdinals.FieldData)
+            ? ((ValuesSource.Bytes.WithOrdinals.FieldData) valuesSource).getIndexFieldName()
+            : null;
     }
 
     String descriptCollectionStrategy() {
         return collectionStrategy.describe();
     }
 
+    public void setWeight(Weight weight) {
+        this.weight = weight;
+    }
+
+    /**
+     Collects term frequencies for a given field from a LeafReaderContext.
+     @param ctx The LeafReaderContext to collect terms from
+     @param globalOrds The SortedSetDocValues for the field's ordinals
+     @param ordCountConsumer A consumer to accept collected term frequencies
+     @return A no-operation LeafBucketCollector implementation, since collection is complete
+     @throws IOException If an I/O error occurs during reading
+     */
+    LeafBucketCollector termDocFreqCollector(
+        LeafReaderContext ctx,
+        SortedSetDocValues globalOrds,
+        BiConsumer<Long, Integer> ordCountConsumer
+    ) throws IOException {
+        if (weight == null || weight.count(ctx) != ctx.reader().maxDoc()) {
+            // Weight not assigned or top-level query does not match all docs in the segment.
+            return null;
+        }
+
+        if (weight.count(ctx) == 0) {
+            // No documents matches top level query on this segment, we can skip the segment
+            return LeafBucketCollector.NO_OP_COLLECTOR;
+        }
+
+        Terms segmentTerms = ctx.reader().terms(this.fieldName);
```

Comment:
I see fieldName can be null when being initialized, do you think it would be dangerous here?

Reply:
+. It's worth checking if the `terms` function implementation would accept a null value.

Reply:
@bowenlan-amzn @ansjcy Thanks for pointing this out. The `terms` function does accept `null` value and returns null in that case.